### PR TITLE
(Engtasks 2800) Adds MSI Build to Cake Recipe

### LINKED ISSUE #3061: MSI Build should be present in this repository
### Checklist

- [X] I have verified this is the correct repository for opening this issue.
- [X] I have verified no other issues exist related to my request.

### Is Your Feature Request Related To A Problem? Please describe.

There is an MSI produced for Chocolatey, but it is not present with the rest of the Chocolatey source code.

### Describe The Solution. Why is it needed?

We want to allow folk to contribute to the MSI, as well as bring the build/release process for the MSI within the main Choco build. This should allow that to happen, as well as opening the doors to producing more MSIs with the Chocolatey Cake Recipe changes required.

### Additional Context

_No response_

### Related Issues

_No response_

### PR DESCRIPTION
**Pull request by JPRuskin**
## Description Of Changes
This adds the MSI creation to the main Chocolatey build process.

## Motivation and Context
We wanted to be able to use a consistent method for producing Chocolatey MSIs. This is an early step in that process, along with the Chocolatey Cake Recipe changes made.

## Testing
1. Checkout this branch
2. Override Chocolatey Cake Recipe content with [104](https://github.com/chocolatey/Chocolatey.Cake.Recipe/pull/104)
3. Build a tagged branch (or remove the condition added to the Build-MSI step within the cake.recipe file)
4. Test produced MSI

### Operating Systems Testing
- Windows 10
- Windows 2019

## Change Types Made

* ~[ ] Bug fix (non-breaking change).~
* [x] Feature / Enhancement (non-breaking change).
* ~[ ] Breaking change (fix or feature that could cause existing functionality to change).~
* ~[ ] Documentation changes.~
* ~[ ] PowerShell code changes.~

## Change Checklist

* ~[ ] Requires a change to the documentation.~
* ~[ ] Documentation has been updated.~
* ~[ ] Tests to cover my changes, have been added.~
* ~[ ] All new and existing tests passed?~
* ~[ ] PowerShell code changes: PowerShell v2 compatibility checked?~

## Related Issue

Fixes #3061

## Notes
Requires release of the Chocolatey Cake Recipe changes before merging.

### CODE REVIEW

File: src/chocolatey.install/assets/Install.ps1 [hunk 1]
```diff
@@ -0,0 +1,683 @@
+<#
```
Comment (gep13) [inline src/chocolatey.install/assets/Install.ps1:1]: Should this file be here?

File: src/chocolatey.install/chocolatey.wxs [hunk 2]
```diff
@@ -0,0 +1,104 @@
...
+    <PropertyRef Id="WIX_IS_NETFRAMEWORK_40_OR_LATER_INSTALLED" />
+    <Condition Message="!(loc.RequiresDotNet4)">
+      <![CDATA[Installed OR WIX_IS_NETFRAMEWORK_40_OR_LATER_INSTALLED]]>
+    </Condition>
+
+    <!-- Custom Actions to Install Chocolatey -->
+    <SetProperty Id="InstallChocolatey"
+      Before="InstallChocolatey"
+      Sequence="execute" 
+      Value="&quot;C:\Windows\System32\WindowsPowerShell\v1.0\powershell.exe&quot; -NoProfile -ExecutionPolicy Bypass -Command &quot;$env:ChocolateyInstall='[INSTALLDIR]'; &amp; '[#ChocolateyInstall]' -ChocolateyDownloadUrl '[#ChocolateyNupkg]' -UseNativeUnzip&quot;" />
```
Comment (JPRuskin) [inline src/chocolatey.install/chocolatey.wxs:39]: We should consider using `%WINDIR%`, or a WiX equivalent.

```suggestion
      Value="&quot;%WINDIR%\System32\WindowsPowerShell\v1.0\powershell.exe&quot; -NoProfile -ExecutionPolicy Bypass -Command &quot;$env:ChocolateyInstall='[INSTALLDIR]'; &amp; '[#ChocolateyInstall]' -ChocolateyDownloadUrl '[#ChocolateyNupkg]' -UseNativeUnzip&quot;" />
```

**Review by gep13 — APPROVED** [review]
LGTM!

**Comment by gep13** [comment]
@JPRuskin thanks for getting this added!

**Pull request merged**